spec/spec_helper.rb

```diff
@@ -18,78 +18,156 @@
   end
 end
 
-def fake_authenticator_data(rp_id: nil, user_present: true, credential_public_key: nil)
-  rp_id ||= "localhost"
-  rp_id_hash = OpenSSL::Digest::SHA256.digest(rp_id)
+class FakeAuthenticator
+  def initialize(creation_options: nil, request_options: nil, context: {})
+    @creation_options = creation_options
+    @request_options = request_options
+    @context = context
```

Comment:
This is truly great work! 👍 

Leaving my 2c, after seeing the authenticator usage, where it's only used for one flow or the other (create or get) I'm kind of wondering if it wouldn't be cleaner to initialize this faker with just one of them at a time. That would lead to having just one options hash and avoid many of the conditionals that have to grab values from either one option hash or the other depending on the request type.

For example in cases like this:
```
      if create?
        creation_options[:challenge] || fake_challenge
      elsif get?
        request_options[:challenge] || fake_challenge
```
you would just have
```
options[:challenge] || fake_challenge
```

Let me know what you think :)

Reply:
Yeah, you're right... those two option hashes feel a bit strange